app/uk/gov/hmrc/lisaapi/controllers/BonusPaymentController.scala

```diff
@@ -88,46 +67,85 @@ class BonusPaymentController extends LisaController with LisaConstants {
       }
     }
 
-  private def processGetBonusPayment(lisaManager:String, accountId:String, transactionId: String)
-                                    (implicit hc: HeaderCarrier, startTime: Long) = {
-    getService.getBonusOrWithdrawal(lisaManager, accountId, transactionId).map {
+  private def processRequestBonusPayment(lisaManager: String, accountId: String, reads: Reads[RequestBonusPaymentRequest])
+                                        (implicit request: Request[AnyContent], startTime: Long, hc: HeaderCarrier) = {
+    withValidJson[RequestBonusPaymentRequest](
+      req => {
+        withValidData(req)(lisaManager, accountId) { () =>
+          postService.requestBonusPayment(lisaManager, accountId, req) map {
+            case success: RequestBonusPaymentSuccessResponse => handleSuccess(lisaManager, accountId, req, success)
+            case failure: RequestBonusPaymentErrorResponse => handleFailure(lisaManager, accountId, req, failure)
+          } recover {
+            case e: Exception => handleError(e, lisaManager, accountId, req)
+          }
+        }
+      },
+      lisaManager = lisaManager
+    )(request, reads, startTime)
+  }
+
+  private def processGetBonusPayment(lisaManager: String, accountId: String, transactionId: String)
+                                    (implicit hc: HeaderCarrier, startTime: Long, request: Request[AnyContent]): Future[Result] = {
+    getService.getBonusOrWithdrawal(lisaManager, accountId, transactionId).flatMap {
       case response: GetBonusResponse =>
         LisaMetrics.incrementMetrics(startTime, OK, LisaMetricKeys.BONUS_PAYMENT)
-        Ok(Json.toJson(response))
+        withApiVersion {
+          case Some(VERSION_1) => {
+            if (response.bonuses.claimReason == "Superseded Bonus") {
+              Logger.warn(s"API v1 received a superseded bonus claim. ID was $transactionId")
+              Future.successful(InternalServerError(Json.toJson(ErrorInternalServerError)))
+            }
+            else {
+              Future.successful(Ok(Json.toJson(response.copy(supersededBy = None))))
+            }
+          }
+          case Some(VERSION_2) => Future.successful(Ok(Json.toJson(response)))
+        }
 
       case _: GetWithdrawalResponse =>
```

Comment:
I believe this case and the subsequent two cases could be merged without any change in functionality - do you think this would make it any easier or harder to understand?

Reply:
I'm not sure they can be merged. Lets review this together.